File: frontend/src/types/enums.ts
/** Mirrors backend `app/core/enums.py` — the single source of this vocabulary
 * on the frontend. Values must stay byte-for-byte identical to the backend. */

export const TaskStatus = {
  TODO: "TODO",
  IN_PROGRESS: "IN_PROGRESS",
  DONE: "DONE",
} as const;
export type TaskStatus = (typeof TaskStatus)[keyof typeof TaskStatus];

export const TaskPriority = {
  LOW: "LOW",
  MEDIUM: "MEDIUM",
  HIGH: "HIGH",
  URGENT: "URGENT",
} as const;
export type TaskPriority = (typeof TaskPriority)[keyof typeof TaskPriority];

export const ProcessingStatus = {
  PENDING: "PENDING",
  PROCESSING: "PROCESSING",
  COMPLETED: "COMPLETED",
  FAILED: "FAILED",
} as const;
export type ProcessingStatus = (typeof ProcessingStatus)[keyof typeof ProcessingStatus];

File: backend/app/core/enums.py
from enum import StrEnum


class TaskStatus(StrEnum):
    """Lifecycle status of a task. Always app-assigned, never LLM-inferred."""

    TODO = "TODO"
    IN_PROGRESS = "IN_PROGRESS"
    DONE = "DONE"


class TaskPriority(StrEnum):
    """Priority inferred by the LLM at extraction time, editable afterwards."""

    LOW = "LOW"
    MEDIUM = "MEDIUM"
    HIGH = "HIGH"
    URGENT = "URGENT"


class ProcessingStatus(StrEnum):
    """Lifecycle status of a meeting's task-extraction pipeline."""

    PENDING = "PENDING"
    PROCESSING = "PROCESSING"
    COMPLETED = "COMPLETED"
    FAILED = "FAILED"


class AIProviderName(StrEnum):
    """Supported AI provider identifiers — the only valid `AI_PROVIDER` values."""

    GEMINI = "gemini"
    GROQ = "groq"
